examples/foreign_contracts.rs

```diff
@@ -326,3 +348,43 @@ pub mod std {
 
     pub mod result {}
 }
+
+pub mod alloc {
+    pub mod vec {
+        pub struct Vec<T> {
+            _phantom: std::marker::PhantomData<T>,
+            len: usize,
+        }
+        
+        impl<T> Vec<T> {
+            pub fn new() -> Vec<T> {
+                Vec {
+                    _phantom: std::marker::PhantomData,
+                    len: 0,
+                }
+            }
+
+            pub fn len(&self) -> usize {
+                self.len
+            }
+
+            pub fn push(&mut self, _value: T) {
+                precondition!(self.len < std::usize::MAX);
+                let old_len = self.len;
+                self.len += 1;
+                verify!(self.len == old_len + 1);
```

Comment:
There is no need to verify this. The previous line already has all the information we need from this contract.